src/middlewared/middlewared/api/base/event.py

```diff
@@ -0,0 +1,21 @@
+from dataclasses import dataclass
+
+from .model import BaseModel
+
+
+@dataclass(slots=True, frozen=True)
```

Comment:
Awesome, but let's also add `kw_only=True`. This will make developers lives easier when they're converting existing events instead of having to memorize position arguments.